### PR TITLE
Upgraded dependencies and refactored

### PR DESCRIPTION
**Pull request by ccw**
Hi @xiaodongw, 

This PR is to upgrade most of the dependencies, and also to remove some hard coded paths. Please see if this make senses to you.

### CODE REVIEW

File: src/main/scala/com/github/xiaodongw/swagger/finatra/WebjarsController.scala [hunk 1]
```diff
@@ -17,17 +17,18 @@ object WebjarsController {
...
+  var resolver: FileResolver = null
+
   private val disableCache: Boolean = false
 
   get(s"${root}/:*") { request: Request =>
     val resourcePath = request.getParam("*")
 
-    val webjarsResourceURI: String = "/META-INF/resources/webjars/" + resourcePath
-    //logger.log(Level.FINE, "Webjars resource requested: {0}", webjarsResourceURI)
+    val webjarsResourceURI: String = s"/META-INF/resources/webjars/swagger-ui/$version/$resourcePath"
```
Comment (xiaodongw) [inline src/main/scala/com/github/xiaodongw/swagger/finatra/WebjarsController.scala]: Before the WebjarsController is general to support all web jars, now it ties to swagger-ui.

File: src/test/scala/com/github/xiaodongw/swagger/finatra/SampleApp.scala [hunk 2]
```diff
@@ -29,7 +29,7 @@ object SampleApp extends HttpServer {
   override def configureHttp(router: HttpRouter) {
     router
       .filter[CommonFilters]
-      .add[WebjarsController]
+      .add(new WebjarsController)
```
Comment (xiaodongw) [inline src/test/scala/com/github/xiaodongw/swagger/finatra/SampleApp.scala]: Have you verified if the sample app still work? I tried it and it can not show the swagger UI.

**Review by xiaodongw — COMMENTED** [review]
The swagger-ui / web jar change seems broken.

File: src/test/scala/com/github/xiaodongw/swagger/finatra/SampleApp.scala (hunk 2 above)
Comment (ccw) [reply src/test/scala/com/github/xiaodongw/swagger/finatra/SampleApp.scala]: Yes, it still work. It is still accessible through 

`http://localhost:8888/api-docs/ui`

or 

`http://localhost:8888/webjars/index.html?url=/api-docs/model`

File: src/main/scala/com/github/xiaodongw/swagger/finatra/WebjarsController.scala [hunk 3]
```diff
@@ -17,17 +17,19 @@ object WebjarsController {
   private val DEFAULT_EXPIRE_TIME_MS: Long = 86400000L // 1 day
 }
 
-class WebjarsController @Inject() (resolver: FileResolver) extends Controller {
+class WebjarsController (jar: String, version: String) extends Controller {
   import WebjarsController._
 
-  private val root: String = "/webjars"
+  @Inject
+  var resolver: FileResolver = _
```
Comment (xiaodongw) [inline src/main/scala/com/github/xiaodongw/swagger/finatra/WebjarsController.scala:24]: Why this field is public? Because of the limitation of injection?

File: src/main/scala/com/github/xiaodongw/swagger/finatra/WebjarsController.scala [hunk 4]
```diff
@@ -17,17 +17,19 @@ object WebjarsController {
...
+
+  private val root: String = "webjars"
```
Comment (xiaodongw) [inline src/main/scala/com/github/xiaodongw/swagger/finatra/WebjarsController.scala]: This is inconsistent, this root is without "/" prefix, but others are with.
I prefer to have "/" prefix for all of them.

File: src/main/scala/com/twitter/finatra/http/SwaggerRouteDSL.scala [hunk 5]
```diff
@@ -11,57 +11,68 @@ object SwaggerRouteDSL {
...
+
   implicit protected val swagger: Swagger
   protected val dsl: RouteDSL
 
   def postWithDoc[RequestType: Manifest, ResponseType: Manifest](route: String, name: String = "", admin: Boolean = false, adminIndexInfo: Option[AdminIndexInfo] = None)
                                                                 (doc: Operation => Unit)
                                                                 (callback: RequestType => ResponseType): Unit = {
-    registerOperation(route, "post")(doc)
-    dsl.post(route, name, admin, adminIndexInfo)(callback)
+    registerOperation(route.tailed, "post")(doc)
```
Comment (xiaodongw) [inline src/main/scala/com/twitter/finatra/http/SwaggerRouteDSL.scala]: What's the reason for this change? Can you give me an example that the old code does not work.

Comment (ccw) [reply src/main/scala/com/twitter/finatra/http/SwaggerRouteDSL.scala]: The changes are mainly to apply the [`basePath`](http://swagger.io/specification/#swaggerBasePath) setting to the registered Finatra routes.

And the `tailed` function is, for my personal convenience, to ensure that there is no consecutive occurrences of `/` in the registered routes, and every route will be tailed with a `/`.  As an example, if the base path is given as `/v1/` and the route is given as `/student`, the `tailed` route will be `/v1/student/` rather than the original `/v1//student`.

To be honest, the `tailed` function is not really necessary and I will remove it.

Comment (xiaodongw) [reply src/main/scala/com/twitter/finatra/http/SwaggerRouteDSL.scala]: This doesn't sound right to me, the purpose of this lib is to provide API document, not changing the behavior of the API. My point is, how I document my API should not change the behavior of the API self.
